pkg/generators/servers.go

```diff
@@ -725,6 +755,10 @@ func (g *ServersGenerator) fieldName(parameter *concepts.Parameter) string {
 	return name
 }
 
+func (g *ServersGenerator) parameterName(parameter *concepts.Parameter) string {
+	return g.names.Private(parameter.Name())
```

Comment:
I think that using `Private` here isn't correct. For a query parameter like `my_parameter` it will generate `myParameter`. Also it will try to avoid conflicts with Go reserved words. That is not what you want. You probably want to do something similar to what we do in the `urlSegment` function.